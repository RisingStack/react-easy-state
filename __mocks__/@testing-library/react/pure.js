module.exports = process.env.NOHOOK
  ? require('no-hook-react-testing-library')
  : require('@testing-library/react/pure');
