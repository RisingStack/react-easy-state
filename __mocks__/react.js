// this is here to avoid duplicate react entries in the examples
// (one from the example's node_modules and one from the root's node_modules)
module.exports = require('react');
