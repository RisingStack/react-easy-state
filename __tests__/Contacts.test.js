import React, { StrictMode } from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react/pure'
import App from '../examples/contacts/src/App'

describe('Contacts App', () => {
  const { container } = render(
    <StrictMode>
      <App />
    </StrictMode>
  )
  afterAll(cleanup)

  test('should add new contacts', () => {
    expect(container).toMatchSnapshot('01. Initial state')

    const nameField = container.querySelector('input[name="name"]')
    const emailField = container.querySelector('input[name="email"]')
    const createButton = container.querySelector('button')

    fireEvent.change(nameField, {
      target: { name: 'name', value: 'Test Contact' }
    })
    expect(container).toMatchSnapshot('02. Create Test Contact name')

    fireEvent.change(emailField, {
      target: { name: 'email', value: 'test.contact@gmail.com' }
    })
    expect(container).toMatchSnapshot('03. Create Test Contact email')

    fireEvent.click(createButton)
    expect(container).toMatchSnapshot('04. Add Test Contact')

    fireEvent.change(nameField, {
      target: { name: 'name', value: '' }
    })
    fireEvent.change(emailField, {
      target: { name: 'email', value: '' }
    })
    fireEvent.click(createButton)
    expect(container).toMatchSnapshot('05. Add Placeholder Contact')
  })

  test('should edit contact', () => {
    console.log('editTEST')
    let display,
      editor,
      editButton,
      cancelButton,
      saveButton,
      nameField,
      emailField

    display = container.querySelector('.contact-display')
    editButton = display.querySelector('.zmdi-edit')

    fireEvent.click(editButton)
    expect(container).toMatchSnapshot('06. Switch Test Contact to Edit Mode')

    editor = container.querySelector('.contact-editor')
    nameField = editor.querySelector('input[name="name"]')
    cancelButton = editor.querySelector('.zmdi-close')

    fireEvent.change(nameField, {
      target: { name: 'name', value: 'Edited Test Contact' }
    })
    expect(container).toMatchSnapshot('07. Edit Test Contact name')

    fireEvent.click(cancelButton)
    expect(container).toMatchSnapshot('08. Cancel Test Contact edit')

    display = container.querySelector('.contact-display')
    editButton = display.querySelector('.zmdi-edit')

    fireEvent.click(editButton)
    expect(container).toMatchSnapshot('09. Switch Test Contact to edit Mode')

    editor = container.querySelector('.contact-editor')
    emailField = editor.querySelector('input[name="email"]')
    saveButton = editor.querySelector('.zmdi-save')

    fireEvent.change(emailField, {
      target: { name: 'email', value: 'test.contact.edited@gmail.com' }
    })
    expect(container).toMatchSnapshot('10. Edit Test Contact email')

    fireEvent.click(saveButton)
    expect(container).toMatchSnapshot('11. Save Test Contact edit')
  })

  test('should delete contact', () => {
    let deleteButton = container.querySelectorAll(
      '.contact-display .zmdi-delete'
    )[1]

    fireEvent.click(deleteButton)
    expect(container).toMatchSnapshot('12. Delete Placeholder Contact')

    deleteButton = container.querySelector('.contact-display .zmdi-delete')

    fireEvent.click(deleteButton)
    expect(container).toMatchSnapshot('13. Delete Test Contact')
  })
})
