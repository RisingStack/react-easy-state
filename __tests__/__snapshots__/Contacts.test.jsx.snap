// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts App should add new contacts: 01. Initial state 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should add new contacts: 02. Create Test Contact name 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value="Test Contact"
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should add new contacts: 03. Create Test Contact email 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value="Test Contact"
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value="test.contact@gmail.com"
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should add new contacts: 04. Add Test Contact 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-display"
      >
        <td>
          Test Contact
        </td>
        <td>
          test.contact@gmail.com
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should add new contacts: 05. Add Placeholder Contact 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-display"
      >
        <td>
          Test Contact
        </td>
        <td>
          test.contact@gmail.com
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-display"
      >
        <td>
          Placeholder
        </td>
        <td>
          Placeholder
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should delete contact: 12. Delete Placeholder Contact 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-display"
      >
        <td>
          Test Contact
        </td>
        <td>
          test.contact.edited@gmail.com
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should delete contact: 13. Delete Test Contact 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should edit contact: 06. Switch Test Contact to Edit Mode 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-editor"
      >
        <td>
          <input
            name="name"
            value="Test Contact"
          />
        </td>
        <td>
          <input
            name="email"
            value="test.contact@gmail.com"
          />
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-save"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-close"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-display"
      >
        <td>
          Placeholder
        </td>
        <td>
          Placeholder
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should edit contact: 07. Edit Test Contact name 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-editor"
      >
        <td>
          <input
            name="name"
            value="Edited Test Contact"
          />
        </td>
        <td>
          <input
            name="email"
            value="test.contact@gmail.com"
          />
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-save"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-close"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-display"
      >
        <td>
          Placeholder
        </td>
        <td>
          Placeholder
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should edit contact: 08. Cancel Test Contact edit 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-display"
      >
        <td>
          Test Contact
        </td>
        <td>
          test.contact@gmail.com
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-display"
      >
        <td>
          Placeholder
        </td>
        <td>
          Placeholder
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should edit contact: 09. Switch Test Contact to edit Mode 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-editor"
      >
        <td>
          <input
            name="name"
            value="Test Contact"
          />
        </td>
        <td>
          <input
            name="email"
            value="test.contact@gmail.com"
          />
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-save"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-close"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-display"
      >
        <td>
          Placeholder
        </td>
        <td>
          Placeholder
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should edit contact: 10. Edit Test Contact email 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-editor"
      >
        <td>
          <input
            name="name"
            value="Test Contact"
          />
        </td>
        <td>
          <input
            name="email"
            value="test.contact.edited@gmail.com"
          />
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-save"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-close"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-display"
      >
        <td>
          Placeholder
        </td>
        <td>
          Placeholder
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contacts App should edit contact: 11. Save Test Contact edit 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="contact-display"
      >
        <td>
          Test Contact
        </td>
        <td>
          test.contact.edited@gmail.com
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-display"
      >
        <td>
          Placeholder
        </td>
        <td>
          Placeholder
        </td>
        <td>
          <button>
            <i
              class="zmdi zmdi-edit"
            />
          </button>
          <button>
            <i
              class="zmdi zmdi-delete"
            />
          </button>
        </td>
      </tr>
      <tr
        class="contact-creator"
      >
        <td>
          <input
            name="name"
            placeholder="Contact name..."
            value=""
          />
        </td>
        <td>
          <input
            name="email"
            placeholder="Contact email..."
            value=""
          />
        </td>
        <td>
          <button>
            Add Contact
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
